keystone/v3.0/combined_membership.js

```diff
@@ -24,24 +24,36 @@ var ProjectMembershipManager = Class.extend({
 
 
   all: function(params, callback) {
-    this._role_assignments.all({
+    var new_params = {
       query: {
-       "scope.project.id": params.data.project
+        "scope.project.id": params.data.project
       }
-    }, _.bind(function(err, assignments) {
+    };
+
+    this._role_assignments.all(new_params, _.bind(function(err, assignments) {
       if (err) {
         this.safe_complete(err, null, null, params, callback);
         return;
       }
 
-      async.map(assignments, _.bind(this._convertAssignment, this, params.data.project), _.bind(function(err, results) {
-        if (err) {
-          this.safe_complete(err, null, null, params, callback);
-          return;
-        }
-
-        this.safe_complete(null, results, { status: 200 }, params, callback);
-      }, this));
+      async.map(assignments,
+        // Iterator
+        _.bind(this._convertAssignment, this, params.data.project),
+
+        // Callback
+        _.bind(function(err, results) {
+          // It's important that we still return a partial data set here even
+          // if some assignments couldn't be mapped correctly. That requires
+          // that _convertAssignment not return an error since async calls the
```

Comment:
This asymmetry in `async.map` is unfortunate, and is really a poor design, since `map` is an inherently unordered/parallelizable concept (as opposed to `each`), yet the bailing behavior and single `err` implies a race. It seems like it should be `map -> xs -> f -> (f -> (errs, results))` instead. :(